lib/pbench/agent/base.py

```diff
@@ -22,16 +22,18 @@ def __init__(self, context):
         self.name = os.path.basename(sys.argv[0])
 
         env_pbench_run = os.environ.get("pbench_run")
-        self.pbench_run = pathlib.Path(
-            env_pbench_run if env_pbench_run else self.config.pbench_run
-        )
-
-        if not self.pbench_run.exists():
-            click.echo(
-                f"[ERROR] pbench run directory does not exist, {self.pbench_run}",
-                err=True,
-            )
-            click.get_current_context().exit(1)
+        if env_pbench_run:
+            self.pbench_run = pathlib.Path(env_pbench_run)
+            if not self.pbench_run.exists():
```

Comment:
To match the `base` script this needs to be `if not self.pbench_run.is_dir()`.

Reply:
Or 

```
    pbench_run = pathlib.Path(env_pbench_run)
    if not pbench_run.is_dir():
        ...
    self.pbench_run = pbench_run
```